src/kernel/MomentumMassElemKernel.C

```diff
@@ -71,14 +71,15 @@ MomentumMassElemKernel<AlgTraits>::MomentumMassElemKernel(
   dataPreReqs.add_cvfem_volume_me(meSCV);
 
   // fields and data
-  dataPreReqs.add_coordinates_field(*coordinates_, AlgTraits::nDim_, CURRENT_COORDINATES);
-  dataPreReqs.add_gathered_nodal_field(*densityNm1_, 1);
-  dataPreReqs.add_gathered_nodal_field(*densityN_, 1);
-  dataPreReqs.add_gathered_nodal_field(*densityNp1_, 1);
-  dataPreReqs.add_gathered_nodal_field(*velocityNm1_, AlgTraits::nDim_);
-  dataPreReqs.add_gathered_nodal_field(*velocityN_, AlgTraits::nDim_);
-  dataPreReqs.add_gathered_nodal_field(*velocityNp1_, AlgTraits::nDim_);
-  dataPreReqs.add_gathered_nodal_field(*Gjp_, AlgTraits::nDim_);
+  const auto& fields = metaData.get_fields();
```

Comment:
@alanw0 I am not crazy about this. Ideally it would be good if `ElemDataRequests` just could take field ordinals. However, it does quite a bit of stuff internally querying about `entity_rank` etc., so I am not sure it would be straightforward. Also doing this keeps the commit smaller and focused on just the conversion of the pointer.

Reply:
Yes we could make ElemDataRequests take field ordinals, if we made ElemDataRequests hold a stk MetaData (it could be a constructor argument to ElemDataRequests.)